src/promptflow/tests/test_configs/flows/script_tool_with_init/script_tool_with_init.py

```diff
@@ -0,0 +1,18 @@
+from jinja2 import Template
+
+from promptflow import ToolProvider, tool
+from promptflow.connections import AzureOpenAIConnection
+from promptflow.contracts.types import PromptTemplate
+
+
+class ScriptToolWithInit(ToolProvider):
+    def __init__(self, connection: AzureOpenAIConnection):
+        super().__init__()
+        self.connection = connection
+
+    @tool
+    def call(self, api: str, prompt: PromptTemplate, **kwargs):
```

Comment:
Let's write a simpler case without connection and prompt template

Reply:
Refined.